lib/experimental/fonts/font-library/font-library.php

```diff
@@ -109,22 +110,8 @@ function gutenberg_init_font_library_routes() {
 	 * @return WP_Font_Collection|WP_Error A font collection is it was registered
 	 *                                     successfully, else WP_Error.
 	 */
-	function wp_register_font_collection( $slug, $args = array() ) {
-		return WP_Font_Library::register_font_collection( $slug, $args );
-	}
-}
-
-if ( ! function_exists( 'wp_register_font_collection_from_json' ) ) {
-	/**
-	 * Registers a new Font Collection from a json file in the Font Library.
-	 *
-	 * @since 6.5.0
-	 *
-	 * @param string $file_or_url File path or URL to a JSON file containing the font collection data.
-	 * @return WP_Font_Collection|WP_Error A font collection if registration was successful, else WP_Error.
-	 */
-	function wp_register_font_collection_from_json( $file_or_url ) {
-		return WP_Font_Library::register_font_collection_from_json( $file_or_url );
+	function wp_register_font_collection( $slug, $args_or_file = array() ) {
```

Comment:
It seems like both arguments should be mandatory now because Wwthout $args_or_file the collection the collection is useless and the data can't be added after the construction of the object.


Currently you can call the function like this:
```
wp_register_font_collection( 'https://dummyjson.com/products/1' );
```
whithout getting any notice about the lack of the right number of parameters.
I think a _doing_it_wrong notice is needed for the the function is called with only one parameter.
Apart from that the collection should not be registered as is happening in this example:

https://github.com/WordPress/gutenberg/pull/58530#issuecomment-1921497428

Reply:
`wp_register_font_collection` and related functions have been updated to require both function parameters.